librz/core/cmd_api.c

```diff
@@ -1089,7 +1099,137 @@ static char *get_help(RzCmd *cmd, RzCmdDesc *cd, RzCmdParsedArgs *args, bool use
 	return NULL;
 }
 
-RZ_API char *rz_cmd_get_help(RzCmd *cmd, RzCmdParsedArgs *args, bool use_color) {
+static void fill_args_json(const RzCmdDesc *cd, PJ *j) {
+	const RzCmdDescArg *arg;
+	bool has_array = false;
+	pj_ka(j, "args");
+	const char *argtype = NULL;
+	for (arg = cd->help->args; arg && arg->name; arg++) {
+		if (has_array) {
+			rz_warn_if_reached();
+			break;
+		}
+		pj_o(j);
+#define CASE_TYPE(x, y) \
+	case (x): \
+		argtype = (y); \
+		break
+		switch (arg->type) {
+			CASE_TYPE(RZ_CMD_ARG_TYPE_FAKE, "fake");
+			CASE_TYPE(RZ_CMD_ARG_TYPE_NUM, "number");
+			CASE_TYPE(RZ_CMD_ARG_TYPE_RZNUM, "expression");
+			CASE_TYPE(RZ_CMD_ARG_TYPE_STRING, "string");
+			CASE_TYPE(RZ_CMD_ARG_TYPE_ENV, "environment_variable");
+			CASE_TYPE(RZ_CMD_ARG_TYPE_ZIGN, "zignature");
+			CASE_TYPE(RZ_CMD_ARG_TYPE_ZIGN_SPACE, "zignature_space");
+			CASE_TYPE(RZ_CMD_ARG_TYPE_CHOICES, "choice");
+			CASE_TYPE(RZ_CMD_ARG_TYPE_FCN, "function");
+			CASE_TYPE(RZ_CMD_ARG_TYPE_FILE, "filename");
+			CASE_TYPE(RZ_CMD_ARG_TYPE_OPTION, "option");
+			CASE_TYPE(RZ_CMD_ARG_TYPE_CMD, "command");
+			CASE_TYPE(RZ_CMD_ARG_TYPE_MACRO, "macro");
+			CASE_TYPE(RZ_CMD_ARG_TYPE_EVAL_KEY, "evaluable");
+			CASE_TYPE(RZ_CMD_ARG_TYPE_EVAL_FULL, "evaluable_full");
+#undef CASE_TYPE
+		default:
+			argtype = "unknown";
+			break;
+		}
+		pj_ks(j, "type", argtype);
+		pj_ks(j, "name", arg->name);
+		if (arg->type == RZ_CMD_ARG_TYPE_FAKE) {
+			pj_end(j);
+			continue;
+		}
+		if (arg->no_space) {
+			pj_kb(j, "nospace", true);
+		}
+		if (!arg->optional) {
+			pj_kb(j, "required", true);
+		}
+		if (arg->flags & RZ_CMD_ARG_FLAG_LAST) {
+			pj_kb(j, "is_last", true);
+		}
+		if (arg->flags & RZ_CMD_ARG_FLAG_ARRAY) {
+			pj_kb(j, "is_array", true);
+		}
+		if (arg->flags & RZ_CMD_ARG_FLAG_OPTION) {
+			pj_kb(j, "is_option", true);
+		}
+		if (arg->default_value) {
+			pj_ks(j, "default", arg->default_value);
+		}
+		if (arg->type == RZ_CMD_ARG_TYPE_CHOICES) {
+			pj_ka(j, "choices");
+			const char **choice = arg->choices;
+			for (; *choice; choice++) {
+				pj_s(j, *choice);
+			}
+			pj_end(j);
+		}
+		pj_end(j);
+	}
+	pj_end(j);
+}
+
+/**
```

Comment:
??

Reply:
it's the doxygen grammar.

Reply:
Yes, i meant that there's a documentation more than necessary.